components/linear_app/sources/common/webhook.mjs

```diff
@@ -0,0 +1,66 @@
+import linearApp from "../../linear_app.app.mjs";
+import constants from "../../common/constants.mjs";
+
+export default {
+  props: {
+    linearApp,
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  methods: {
+    setWebhookId(id) {
+      this.db.set(constants.WEBHOOK_ID, id);
+    },
+    getWebhookId() {
+      return this.db.get(constants.WEBHOOK_ID);
+    },
+    isRelevant(body) {
+      return this.getActions().includes(body?.action);
+    },
+    isWebhookValid(clientIp) {
+      return constants.CLIENT_IPS.includes(clientIp);
+    },
+  },
```

Comment:
We should include any methods that are called in `webhook.mjs` just in case a component is created that doesn't implement one of them.
```suggestion
      return constants.CLIENT_IPS.includes(clientIp);
    },
    getResourceTypes() {
      throw new Error("getResourceTypes is not implemented");
    },
    getWebhookLabel() {
      throw new Error("getWebhookLabel is not implemented");
    },
    getActions() {
      throw new Error("getActions is not implemented");
    },
    getMetadata() {
      throw new Error("getMetadata is not implemented");
    },
  },
```